tableau_utilities/scripts/connection.py

```diff
@@ -164,10 +173,16 @@ def connection(args, server=None):
         if debugging_logs:
             print(f'EMBEDDING CREDS FOR DATASOURCE ID: {datasource_id}')
 
-        if datasource_id is None:
-            datasource_id = server.get_datasource(datasource_name=args.name, datasource_project=args.project_name).id
+        obj = getattr(server, f'get_datasource')(object_id, object_name, project_name)
```

Comment:
Also, shouldn't this fail since object_id, object_name, and project_name are now set after this function is called?

Reply:
@JustinGrilli  I thought that the `getattr` did the logic for dealing with whether a user entered either an `id` or a `name` and `project_name` combo?

Reply:
`getattr` is a built-in Python function that will return the attribute of a class.
In this case the attribute is the function `get_datasource`.
The reason it is useful, is because we may want to get a different function, based on user inputs from the CLI (like we do for get_datasources vs get_workbooks in other areas of the codebase).